src/main/java/org/codehaus/mojo/versions/ordering/MajorMinorIncrementalFilter.java

```diff
@@ -106,22 +106,11 @@ else if ( artifactVersion.getIncrementalVersion() != selectedVersion.getIncremen
                     }
                 }
             }
-            else if(selectedVersion.getQualifier() != null){
-                if(selectedVersion.getQualifier().contentEquals("SNAPSHOT")){
-                    if ( !versionsToUse.contains( artifactVersion ) )
-                    {
-                            versionsToUse.add( artifactVersion );
-                    }
-                }
-            }
-            else
-            {
-                if ( allowMajorUpdates && allowMinorUpdates && allowIncrementalUpdates )
```

Comment:
we do not need to check these, if the major minor or incremental version had changed we would not reach this `else`  thus what we have left is a qualifier or build number